Source/FieldSolver/FiniteDifferenceSolver/FiniteDifferenceAlgorithms/CartesianNodalAlgorithm.H

```diff
@@ -37,6 +39,17 @@ struct CartesianNodalAlgorithm {
         stencil_coefs_z[0] = 1._rt/cell_size[2];
     }
 
+    /**
+     * Compute the maximum timestep, for which the scheme remains stable
+     * (Courant-Friedrichs-Levy limit) */
+    static amrex::Real ComputeMaxDt ( amrex::Real const * const dx ) {
+        amrex::Real deltat  = 1./( std::sqrt( AMREX_D_TERM(
+                                       1./(dx[0]*dx[0]),
+                                     + 1./(dx[1]*dx[1]),
+                                     + 1./(dx[2]*dx[2]) )) * PhysConst::c );
+        return deltat;
```

Comment:
Calculating the dt in SP instead of DP with final DP cast:
```suggestion
        amrex::Real const delta_t  = 1._rt / ( std::sqrt( AMREX_D_TERM(
                                           1._rt/(dx[0]*dx[0]),
                                         + 1._rt/(dx[1]*dx[1]),
                                         + 1._rt/(dx[2]*dx[2])
                                     ) ) * PhysConst::c );
        return delta_t;
```
The macro magic with the `+` is strong in this one :)